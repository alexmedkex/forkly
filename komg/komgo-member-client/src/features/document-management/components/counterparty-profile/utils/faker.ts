import { CounterpartyProfile, RiskLevel } from '../../../../counterparties/store/types'

const defaultProfile: CounterpartyProfile = {
  id: 'anon-profile-id',
  counterpartyId: 'anon-counterparty-id',
  renewalDate: new Date('2019-07-16').toISOString(),
  riskLevel: RiskLevel.unspecified,
  managedById: 'anon-user-id'
}

export const fakeProfile = (partial?: Partial<CounterpartyProfile>) => {
  return partial ? { ...defaultProfile, ...partial } : defaultProfile
}

export const mockDate = () => {
  const RealDate = Date
  const _GLOBAL: any = global // we love typescript!

  return {
    freeze: (isoDate: string | number | Date) => {
      _GLOBAL.Date = class extends RealDate {
        constructor(...args: any) {
          super()
          return new RealDate(isoDate)
        }
      }
    },
    restore: () => {
      global.Date = RealDate
    }
  }
}
