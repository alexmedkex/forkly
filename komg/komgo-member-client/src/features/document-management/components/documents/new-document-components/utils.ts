import { HasId, HasName } from '../../../store/types'

export const hasIdAndNameToOption = (collection: Array<HasId & HasName>): Array<{ key: string }> => {
  const originalCollection = collection.map(el => {
    return { key: el.id, text: el.name, value: el.id }
  })
  return originalCollection.reduce((unique: Array<{ key: string }>, o: { key: string }) => {
    if (!unique.some((obj: { key: string }) => obj.key === o.key)) {
      unique.push(o)
    }
    return unique
  }, [])
}
