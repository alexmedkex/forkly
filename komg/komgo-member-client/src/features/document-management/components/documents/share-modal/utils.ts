import { ICompanyTableItemWithSharedDocs, ICounterpartyWithDocuments } from './SelectedCounterparties'
import { FlagNameValues } from 'semantic-ui-react'
import * as i18nIsoCountries from 'i18n-iso-countries'

export function pluralize(word: string, count: number, singleSuffix: string, pluralSuffix: string) {
  if (count > 1) {
    return `${word}${pluralSuffix}`
  }

  return `${word}${singleSuffix}`
}

export const buildCounterpartyWithDocumentsPickerItems = (
  counterpartiesWithDocs: ICounterpartyWithDocuments[]
): ICompanyTableItemWithSharedDocs[] => {
  i18nIsoCountries.registerLocale(require('i18n-iso-countries/langs/en.json'))
  return counterpartiesWithDocs.map(item => ({
    name: item.counterparty.x500Name.CN,
    countryName: i18nIsoCountries.getName(item.counterparty.x500Name.C.trim(), 'en'),
    country: item.counterparty.x500Name.C.toLocaleLowerCase().trim() as FlagNameValues,
    location: item.counterparty.x500Name.L,
    id: item.counterparty.staticId,
    sharedDocuments: item.documents
  }))
}
