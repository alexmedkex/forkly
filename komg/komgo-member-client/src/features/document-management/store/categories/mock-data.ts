import { Category } from '../types'
import { mockProduct } from '../products/mock-data'

export const mockCategories: Category[] = [
  {
    id: 'company-details',
    product: mockProduct,
    name: 'Company Details'
  },
  {
    id: 'management-and-directors',
    product: mockProduct,
    name: 'Management and directors'
  },
  {
    id: 'shareholders',
    product: mockProduct,
    name: 'Shareholders - Ultimate Beneficiary Owners (UBOs)'
  },
  {
    id: 'business-description',
    product: mockProduct,
    name: 'Business description'
  },
  {
    id: 'regulation-and-compliance',
    product: mockProduct,
    name: 'Regulation/Compliance'
  },
  {
    id: 'banking-documents',
    product: mockProduct,
    name: 'Banking documents'
  },
  {
    id: 'miscellaneous',
    product: mockProduct,
    name: 'Miscellaneous'
  }
]
