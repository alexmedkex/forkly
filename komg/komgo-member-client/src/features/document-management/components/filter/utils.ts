import { Category, DocumentType } from '../../index'
import { sortCategories, sortDocumentTypes } from '../../utils/sortingHelper'
import { categoryToColor } from '../documents/document-library/categoryToColor'

export const getDocumentTypeFilterOptions = (categories: Category[], types: DocumentType[]) => {
  return sortCategories(categories).map(c => ({
    name: c.name,
    value: c.id,
    display: {
      color: categoryToColor[c.id]
    },
    items: sortDocumentTypes(types)
      .filter(t => t.category.id === c.id)
      .map(t => ({
        name: t.name,
        value: t.id
      }))
  }))
}
