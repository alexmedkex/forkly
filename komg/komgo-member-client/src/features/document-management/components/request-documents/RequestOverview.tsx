import * as React from 'react'
import styled from 'styled-components'
import moment from 'moment'

import { SectionCard } from './SectionCard'
import RequestInfoLabel from './RequestInfoLabel'
import { Counterparty } from '../../../counterparties/store/types'
import { Request } from '../../store'
import { displayDate } from '../../../../utils/date'
import { getCompanyName } from '../../../counterparties/utils/selectors'
import { SPACES } from '@komgo/ui-components'
import { dark } from '../../../../styles/colors'

export enum RequestSide {
  Receiver = 'Receiver',
  Sender = 'Sender'
}

interface IProps {
  request: Request
  counterparty: Counterparty
  requestSide: RequestSide
}

const RequestOverview: React.FC<IProps> = (props: IProps) => {
  const timeZone = moment.tz(moment.tz.guess()).zoneAbbr()

  const dateOrDash = ({ date, format = 'DD MMM YYYY [-] h:mm A', withTimeZone = true }) => {
    if (date) {
      const formatedDate = displayDate(date, format)
      return withTimeZone ? `${formatedDate} ${timeZone}` : formatedDate
    }
    return '-'
  }

  return (
    <SectionCard title="REQUEST OVERVIEW">
      <RequestOverviewWrapper>
        <div>
          <RequestInfoLabel label="Counterparty">
            <div data-test-id="counteparty-name">{getCompanyName(props.counterparty)}</div>
          </RequestInfoLabel>
          <RequestInfoLabel label={props.requestSide === RequestSide.Sender ? 'Sent on' : 'Received on'}>
            <div data-test-id={props.requestSide === RequestSide.Sender ? 'sent-date' : 'received-date'}>
              {dateOrDash({ date: props.request.createdAt })}
            </div>
          </RequestInfoLabel>
          <RequestInfoLabel label="Due date">
            <div data-test-id="due-date">
              {dateOrDash({ date: props.request.deadline, format: 'D MMM YYYY', withTimeZone: false })}
            </div>
          </RequestInfoLabel>
          <RequestInfoLabel label="Last updated">
            <div data-test-id="last-updated-date">
              {dateOrDash({ date: props.request.updatedAt || props.request.createdAt })}
            </div>
          </RequestInfoLabel>
        </div>
        {/* This will be implemented later <div>Placeholder for right side</div> */}
      </RequestOverviewWrapper>
    </SectionCard>
  )
}

const RequestOverviewWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  color: ${dark};
`

export default RequestOverview
