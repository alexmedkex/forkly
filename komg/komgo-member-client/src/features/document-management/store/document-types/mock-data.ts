import { DocumentType, Field } from '../types'
import { mockProduct } from '../products/mock-data'
import { mockCategories } from '../categories/mock-data'

const defaultFields: Field[] = [
  {
    id: 'createdBy',
    name: 'createdBy',
    type: 'string',
    isArray: false
  },
  {
    id: 'createdOn',
    name: 'createdOn',
    type: 'date',
    isArray: false
  },
  {
    id: 'value',
    name: 'value',
    type: 'string',
    isArray: false
  },
  {
    id: 'expiresOn',
    name: 'expiry-date',
    type: 'date',
    isArray: false
  }
]

const anonDocumentType1: DocumentType = {
  product: mockProduct,
  category: mockCategories[0],
  name: 'Certificate of Incorporation',
  id: 'certificate-of-incorporation',
  fields: defaultFields,
  predefined: false
}

const anonDocumentType2: DocumentType = {
  product: mockProduct,
  category: mockCategories[1],
  name: 'List of Directors',
  id: 'list-of-directors',
  fields: defaultFields,
  predefined: false
}

const anonDocumentType3: DocumentType = {
  product: mockProduct,
  category: mockCategories[2],
  name: 'Passports of UBOs',
  id: 'passports-of-ubos',
  fields: defaultFields,
  predefined: false
}

export const mockDocumentTypes: DocumentType[] = [anonDocumentType1, anonDocumentType2, anonDocumentType3]
