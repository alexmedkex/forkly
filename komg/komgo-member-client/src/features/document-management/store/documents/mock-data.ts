import { Document } from '../types'
import { mockProduct } from '../products/mock-data'
import { mockCategories } from '../categories/mock-data'
import { mockDocumentTypes } from '../document-types/mock-data'
import { fakeDocument } from '../../utils/faker'

const recentRandomDate = (minutesAgo: number): Date => {
  const now = new Date().getTime()
  return new Date(now - minutesAgo)
}

const mockedDate = new Date('December 17, 1995 03:24:00')

const MOCK_COMPANY_ID = 'company-id'

const anonDocument1: Document = fakeDocument({
  id: '1',
  documentId: 'document-id',
  name: 'AML Letter - test',
  product: mockProduct,
  category: mockCategories[0],
  type: mockDocumentTypes[0],
  owner: {
    firstName: 'Owner',
    lastName: 'Owner',
    companyId: MOCK_COMPANY_ID
  },
  hash: 'hash',
  registrationDate: recentRandomDate(Math.random() * 1000 * 3600),
  metadata: [],
  content: undefined,
  sharedWith: [{ counterpartyId: 'societegenerale', sharedDates: [mockedDate] }],
  sharedBy: ''
})

const anonDocument2: Document = fakeDocument({
  id: '2',
  documentId: 'document-id-2',
  name: 'Bank Lines - test',
  product: mockProduct,
  category: mockCategories[1],
  type: mockDocumentTypes[1],
  owner: {
    firstName: 'Owner',
    lastName: 'Owner',
    companyId: MOCK_COMPANY_ID
  },
  hash: 'hash',
  registrationDate: recentRandomDate(Math.random() * 1000 * 3600),
  metadata: [],
  content: undefined,
  sharedWith: [
    { counterpartyId: 'societegenerale', sharedDates: [recentRandomDate(Math.random() * 1000 * 3600)] },
    { counterpartyId: 'bnpparibas', sharedDates: [recentRandomDate(Math.random() * 1000 * 3600)] }
  ],
  sharedBy: ''
})

const anonDocument3: Document = fakeDocument({
  id: '3',
  documentId: 'document-id-3',
  name: 'Board Resolution - test',
  product: mockProduct,
  category: mockCategories[2],
  type: mockDocumentTypes[2],
  owner: {
    firstName: 'Owner',
    lastName: 'Owner',
    companyId: MOCK_COMPANY_ID
  },
  hash: 'hash',
  registrationDate: recentRandomDate(Math.random() * 1000 * 3600),
  metadata: [],
  content: undefined,
  sharedWith: [],
  sharedBy: ''
})

export const anonDocument4: Document = fakeDocument({
  id: '3',
  documentId: 'document-id-4',
  name: 'Board Resolution - test.pdf',
  product: mockProduct,
  category: mockCategories[2],
  type: mockDocumentTypes[2],
  owner: {
    firstName: 'Owner',
    lastName: 'Owner',
    companyId: MOCK_COMPANY_ID
  },
  hash: 'hash',
  registrationDate: recentRandomDate(Math.random() * 1000 * 3600),
  metadata: [],
  content: undefined,
  sharedWith: [],
  sharedBy: ''
})

export const mockDocuments: Document[] = [anonDocument1, anonDocument2, anonDocument3]
