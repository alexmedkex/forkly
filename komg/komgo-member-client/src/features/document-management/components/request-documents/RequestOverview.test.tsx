import * as React from 'react'
import * as renderer from 'react-test-renderer'
import moment from 'moment-timezone'

import RequestOverview, { RequestSide } from './RequestOverview'
import { fakeCounterparty } from '../../../letter-of-credit-legacy/utils/faker'
import { mockRequest } from '../../store/requests/mock-data'

describe('RequestOverview', () => {
  const defaultProps = {
    request: { ...mockRequest[0], createdAt: '2019-08-12T16:10:31.811Z', updatedAt: '2019-08-12T16:10:31.811Z' },
    counterparty: fakeCounterparty(),
    requestSide: RequestSide.Sender
  }

  beforeEach(() => {
    moment.tz.guess = jest.fn(() => 'Europe/Belgrade')
    moment.tz.setDefault('Europe/Belgrade')
    Date.now = jest.fn(() => 1487076708000)
  })

  it('should match sender snapshot', () => {
    expect(renderer.create(<RequestOverview {...defaultProps} />).toJSON()).toMatchSnapshot()
  })

  it('should match receiver snapshot', () => {
    expect(
      renderer.create(<RequestOverview {...defaultProps} requestSide={RequestSide.Receiver} />).toJSON()
    ).toMatchSnapshot()
  })

  it('should match snapshot with - when dates does not exists', () => {
    expect(renderer.create(<RequestOverview {...defaultProps} request={mockRequest[0]} />).toJSON()).toMatchSnapshot()
  })
})
