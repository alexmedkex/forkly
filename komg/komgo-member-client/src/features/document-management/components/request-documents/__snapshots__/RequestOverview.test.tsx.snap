// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestOverview should match receiver snapshot 1`] = `
.c1.c1.c1.c1.c1.c1.c1.c1.c1.c1.c1 {
  margin: 0px;
  padding: 16px 30px 30px 30px;
}

.c0.c0.c0.c0.c0.c0.c0.c0.c0.c0.c0 {
  height: 325px;
  width: unset;
  margin: 30px;
  padding-left: 0px;
  padding-rigth: 0px;
  -webkit-box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
  -moz-box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
  box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
}

.c2.c2.c2.c2.c2.c2.c2.c2.c2.c2.c2 {
  font-size: 11px;
  height: 21px;
  width: 549px;
  color: #1c2936;
  font-weight: 600;
  line-height: 21px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-bottom: 24px;
  padding: 8px 0;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
}

.c6 {
  line-height: 21px;
}

.c5 {
  display: inline-block;
  color: #5d768f;
  text-transform: uppercase;
  font-size: 11px;
  font-weight: 600;
  min-width: 150px;
  line-height: 21px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  color: #1c2936;
}

<div
  className="ui card c0"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="content c1"
  >
    <div
      className="header c2"
    >
      REQUEST OVERVIEW
    </div>
    <div
      className="description "
    >
      <div
        className="c3"
      >
        <div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Counterparty
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="counteparty-name"
              >
                Applicant Name
              </div>
            </div>
          </div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Received on
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="received-date"
              >
                12 Aug 2019 - 6:10 PM CET
              </div>
            </div>
          </div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Due date
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="due-date"
              >
                -
              </div>
            </div>
          </div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Last updated
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="last-updated-date"
              >
                12 Aug 2019 - 6:10 PM CET
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RequestOverview should match sender snapshot 1`] = `
.c1.c1.c1.c1.c1.c1.c1.c1.c1.c1.c1 {
  margin: 0px;
  padding: 16px 30px 30px 30px;
}

.c0.c0.c0.c0.c0.c0.c0.c0.c0.c0.c0 {
  height: 325px;
  width: unset;
  margin: 30px;
  padding-left: 0px;
  padding-rigth: 0px;
  -webkit-box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
  -moz-box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
  box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
}

.c2.c2.c2.c2.c2.c2.c2.c2.c2.c2.c2 {
  font-size: 11px;
  height: 21px;
  width: 549px;
  color: #1c2936;
  font-weight: 600;
  line-height: 21px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-bottom: 24px;
  padding: 8px 0;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
}

.c6 {
  line-height: 21px;
}

.c5 {
  display: inline-block;
  color: #5d768f;
  text-transform: uppercase;
  font-size: 11px;
  font-weight: 600;
  min-width: 150px;
  line-height: 21px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  color: #1c2936;
}

<div
  className="ui card c0"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="content c1"
  >
    <div
      className="header c2"
    >
      REQUEST OVERVIEW
    </div>
    <div
      className="description "
    >
      <div
        className="c3"
      >
        <div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Counterparty
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="counteparty-name"
              >
                Applicant Name
              </div>
            </div>
          </div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Sent on
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="sent-date"
              >
                12 Aug 2019 - 6:10 PM CET
              </div>
            </div>
          </div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Due date
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="due-date"
              >
                -
              </div>
            </div>
          </div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Last updated
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="last-updated-date"
              >
                12 Aug 2019 - 6:10 PM CET
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RequestOverview should match snapshot with - when dates does not exists 1`] = `
.c1.c1.c1.c1.c1.c1.c1.c1.c1.c1.c1 {
  margin: 0px;
  padding: 16px 30px 30px 30px;
}

.c0.c0.c0.c0.c0.c0.c0.c0.c0.c0.c0 {
  height: 325px;
  width: unset;
  margin: 30px;
  padding-left: 0px;
  padding-rigth: 0px;
  -webkit-box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
  -moz-box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
  box-shadow: 0px 1px 11px -4px rgba(0,0,0,0.75);
}

.c2.c2.c2.c2.c2.c2.c2.c2.c2.c2.c2 {
  font-size: 11px;
  height: 21px;
  width: 549px;
  color: #1c2936;
  font-weight: 600;
  line-height: 21px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-bottom: 24px;
  padding: 8px 0;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
}

.c6 {
  line-height: 21px;
}

.c5 {
  display: inline-block;
  color: #5d768f;
  text-transform: uppercase;
  font-size: 11px;
  font-weight: 600;
  min-width: 150px;
  line-height: 21px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  color: #1c2936;
}

<div
  className="ui card c0"
  onClick={[Function]}
  style={Object {}}
>
  <div
    className="content c1"
  >
    <div
      className="header c2"
    >
      REQUEST OVERVIEW
    </div>
    <div
      className="description "
    >
      <div
        className="c3"
      >
        <div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Counterparty
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="counteparty-name"
              >
                Applicant Name
              </div>
            </div>
          </div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Sent on
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="sent-date"
              >
                -
              </div>
            </div>
          </div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Due date
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="due-date"
              >
                -
              </div>
            </div>
          </div>
          <div
            className="c4"
          >
            <span
              className="c5"
            >
              Last updated
            </span>
            <div
              className="c6"
            >
              <div
                data-test-id="last-updated-date"
              >
                -
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
